From f35af2704c6e8a1552b0df37f69afeb4265a6c4b Mon Sep 17 00:00:00 2001
From: Adrian Negreanu <adrian.negreanu@nxp.com>
Date: Thu, 8 Jun 2017 18:08:03 +0300
Subject: [PATCH] MGS-2963 [#ccc] Miss /usr/bin/eglretrace file in FB and XWLD
 rootfs.

Not yet upstreamable since the NonX backend is not yet upstream.

Upstream-Status: Inappropriate [other]
Signed-off-by: Adrian Negreanu <adrian.negreanu@nxp.com>
---
 retrace/glws_nonx.cpp | 29 +++++++++++++++++++++++++----
 1 file changed, 25 insertions(+), 4 deletions(-)

diff --git a/retrace/glws_nonx.cpp b/retrace/glws_nonx.cpp
index fa8db97..18df45a 100644
--- a/retrace/glws_nonx.cpp
+++ b/retrace/glws_nonx.cpp
@@ -74,8 +74,8 @@ public:
     EGLNativePixmapType native_pixmap;
     EGLenum api;
 
-    NonxDrawable(const Visual *vis, int w, int h, bool pbuffer) :
-        Drawable (vis, w, h, pbuffer),
+    NonxDrawable(const Visual *vis, int w, int h, const pbuffer_info *info) :
+        Drawable (vis, w, h, info),
         api(EGL_OPENGL_ES_API)
     {
         native_pixmap = (EGLNativePixmapType)0;
@@ -387,9 +387,10 @@ createVisual(bool doubleBuffer, unsigned samples, Profile profile) {
 }
 
 Drawable *
-createDrawable(const Visual *visual, int width, int height, bool pbuffer)
+createDrawable(const Visual *visual, int width, int height,
+        const pbuffer_info *info)
 {
-    return new NonxDrawable(visual, width, height, pbuffer);
+    return new NonxDrawable(visual, width, height, info);
 }
 
 Context *
@@ -492,4 +493,24 @@ processEvents(void)
     return false;
 }
 
+bool
+bindTexImage(Drawable *pBuffer, int iBuffer) {
+    std::cerr << "error: NonX::wglBindTexImageARB not implemented.\n";
+    assert(pBuffer->pbuffer);
+    return true;
+}
+
+bool
+releaseTexImage(Drawable *pBuffer, int iBuffer) {
+    std::cerr << "error: NonX::wglReleaseTexImageARB not implemented.\n";
+    assert(pBuffer->pbuffer);
+    return true;
+}
+
+bool
+setPbufferAttrib(Drawable *pBuffer, const int *attribList) {
+    assert(pBuffer->pbuffer);
+    return true;
+}
+
 }
-- 
2.7.4

